module2/lessons/how_the_web_works_api.md

```diff
@@ -0,0 +1,219 @@
+
+Requests are created by:
+
+1. Typing a URL into a browser address bar
+   - Always a `GET` request
+   - Path is hand-typed
+   - Can include Query Params that must be hand-typed
+   - Does not include a body
+1. Clicking Links or Buttons
+   - The developer decides what Verb, Path and Query Params are included
+   - Does not include a body
+1. Form submission
+   - Developer decides on Verb and Path
+   - Can include query params but usually does not.
+   - User inputs data into form fields. This data is included in the body of the request
+
+## Introduction to APIs
+
+An API (Application Programming Interface) is a set of rules and protocols that allows different software applications to communicate with each other. In the context of web development, an API often refers to a web server that accepts requests and sends data responses, typically in JSON format, instead of HTML.
+
+Frontend applications call APIs all the time to get information to display to users.
+
+For example, when you see a weather widget on a website, that frontend is requesting data from a weather API. That weather API is just sending back raw JSON data, and it is up to the client application to display it.
+
+![Weather Widget API call](../../assets/images/lessons/how_the_web_works_api/weather_example.png)
+
+### Key Concepts of Web APIs
+
+1. **Endpoints**: Specific URLs that represent different resources or actions, typically used to retrieve data
+2. **Request Methods**: HTTP verbs that indicate the desired action
+3. **Request/Response Format**: Often JSON for modern web APIs
+
+## Introduction to JSON
+
+JSON (JavaScript Object Notation) is a lightweight data interchange format that is easy for humans to read and write and easy for machines to parse and generate.
+
+```json
+{
+  "name": "John Doe",
+  "age": 30,
+  "city": "New York",
+  "is_member": false,
+  "hobbies": ["coding", "gaming", "reading"]
+}
+```
+
+## Practicing API Requests with Postman
+
+Another common way to call an API is by using an application such as Postman. Postman is a popular tool for testing API requests. Let's practice making requests to a sample API:
+
+1. Install Postman if you haven't already
+2. We'll use the [Github API](https://docs.github.com/en/rest?apiVersion=2022-11-28) for practice.
+3. Try the following requests:
+   - GET `https://api.github.com/users`
+   - GET `https://api.github.com/users/<your_username>`
+
+When thinking about the role that APIs play in our web applications, they are another data source for us to call on to retrieve some data, similar to accessing data from your own database. The biggest difference being, you don't have to manage the API database yourself. How cool is that!
+
+<section class="call-to-action">
+### What are some potential drawbacks of using an API?
```

Comment:
I'd love to see this be a drop down with bullet points inside of what some of those drawbacks are. I feel like our lessons should be fully built out so that a student could read them and not be left wondering what the answer is. :) IMO

Reply:
I was thinking of this more like a turn-and-talk. I could also add some examples here as well.

Reply:
@KatBrandt 

> our lessons should be fully built out so that a student could read them and not be left wondering what the answer is

I agree with this.  As open source curriculum, I think this is a really valuable way to approach our lessons.